src/helpers/check-merge-safety.ts

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2021 Expedia, Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    https://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { HelperInputs } from '../types/generated';
+import { context } from '@actions/github';
+import { octokit } from '../octokit';
+import micromatch from 'micromatch';
+import { PullRequest } from '../types/github';
+import { paginateAllOpenPullRequests } from '../utils/paginate-open-pull-requests';
+import { map } from 'bluebird';
+import { setCommitStatus } from './set-commit-status';
+
+export class CheckMergeSafety extends HelperInputs {
+  paths?: string;
+  override_filter_paths?: string;
+  override_filter_globs?: string;
+}
+
+export const checkMergeSafety = async (inputs: CheckMergeSafety) => {
+  const isPrWorkflow = Boolean(context.issue.number);
+  if (!isPrWorkflow) {
+    return handlePushWorkflow(inputs);
+  }
+  const { data: pullRequest } = await octokit.pulls.get({ pull_number: context.issue.number, ...context.repo });
+
+  const message = await getMergeSafetyMessage(pullRequest, inputs);
+
+  if (message) {
+    throw new Error(message);
+  }
+};
```

Comment:
In the `pull_request` workflow, we don't need to set a status because it's the actual job running. Just gotta make sure the job name matches the status

Reply:
cool makes sense